### PR TITLE
[Spring App] Support OpenAPI URI in spring cloud gateway route config

### PR DESCRIPTION
**Pull request by ninpan-ms**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Spring App

File: src/spring/azext_spring/gateway.py [hunk 1]
```diff
@@ -207,26 +202,38 @@ def _validate_route_config_exist(client, resource_group, service, name):
...
+    if app_resource_id is not None:
+        route_properties.app_resource_id = app_resource_id
+    route_config_resource = models.GatewayRouteConfigResource(
+        properties=route_properties)
+    return client.gateway_route_configs.begin_create_or_update(resource_group, service, DEFAULT_NAME, name, route_config_resource)
+
+
+def _get_app_resource_id_by_name(client, resource_group, service, app_name):
     if app_name is not None:
         app_resource = client.apps.get(resource_group, service, app_name)
```
Comment (yuwzho) [inline src/spring/azext_spring/gateway.py:219]: NIT: this will result an additional network call, how about spelling the resource id in the client side?

Comment (ninpan-ms) [reply src/spring/azext_spring/gateway.py:219]: This is not only to construct the app resource Id, but also used to check if the app exist.

**Comment by ninpan-ms** [comment]
Please do not merge until backend release ready around 06-21. @zhoxing-ms but you can help review it first, thanks!

File: src/spring/azext_spring/gateway.py (hunk 1 above)
Comment (ninpan-ms) [reply src/spring/azext_spring/gateway.py:219]: This is not only to construct the app resource Id, but also used to check if the app exist.

**Comment by zhoxing-ms** [comment]
@ninpan-ms This PR looks good to me. I add a label `do-not-merge` on it first. Please remind me when we need merge it, thank you

**Review by yuwzho — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**